### PR TITLE
chore: add-error-on-unused-import

### PR DESCRIPTION
**Pull request by Tuditi**
## Summary

Update rules on so that linter doesn't allow unused variables and imports.

### Changelog
```
- fixed @iota/wallet to 1.0.12
- enabled no-unused-vars rule on typescript-eslint
```

## Relevant Issues

https://github.com/iotaledger/firefly/issues/3283

## Type of Change
Please select any type below that is applicable to your changes, and delete those that are not.
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [x] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [ ] __New__ - a change which implements a new feature
- [ ] __Update__ - a change which updates existing functionality

### Platforms
Please select any platforms where your changes have been tested.
- __Desktop__
	- [ ] MacOS
	- [x] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions
Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

## Checklist
Please tick all of the following boxes that are relevant to your changes, and delete those that are not.
- [x] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Code looks pretty good! Just some minor things.

File: packages/shared/components/inputs/Pin.svelte [hunk 1]
```diff
@@ -53,27 +53,26 @@
...
-        } else if (e.keyCode === KEYBOARD.ENTER) {
+        } else if (event.key === KEYBOARD.ENTER) {
             if (validatePinFormat(inputs.join(''))) {
                 dispatch('submit')
             }
-        } else if (e.keyCode === KEYBOARD.TAB) {
+        } else if (event.key === KEYBOARD.TAB) {
             // Do default tab handling by focusing the root
             // container and allow default processing to happen
             root.focus()
```
Comment (rajivshah3) [inline packages/shared/components/inputs/Pin.svelte:67]: Should this be changed in a separate PR? Good call on switching to `key` btw — I had been meaning to change that and then I forgot 😅

Comment (Tuditi) [reply packages/shared/components/inputs/Pin.svelte:67]: I think it's fine like this. Imo it's not worth of creating a separate PR. There are 2 other files which have `keyCode`, I'll update them aswell and put it in the change description of this PR.

File: packages/mobile/capacitor/capacitorApi.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ let activeProfileId = null
 export const nativeSplash = SplashScreen
 
 export const CapacitorApi: IPlatform = {
```
[the remark this answers is not in the document]
Comment (Tuditi) [reply packages/mobile/capacitor/capacitorApi.ts:21]: I don't think so. We can write those methods in the future imo.

File: packages/shared/lib/profile.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import { AccountColors } from '@core/account'
-import { IProfile } from '@core/profile'
-import { get } from 'svelte/store'
+// import { AccountColors } from '@core/account'
```
Comment (MarkNerdi) [inline packages/shared/lib/profile.ts:1]:
```
ERROR in /home/mark/Work/firefly/packages/shared/lib/profile.ts(1,1)
      TS1208: All files must be modules when the '--isolatedModules' flag is provided.
```

**Review by MarkNerdi — COMMENTED** [review]
I get one error while compiling.

File: packages/shared/components/inputs/Pin.svelte (hunk 1 above)
Comment (rajivshah3) [reply packages/shared/components/inputs/Pin.svelte:67]: IMO it should be in a separate PR because it can affect the functionality of the PIN component (making it potentially user-facing, and no longer considered a chore). We've seen reports of issues with other keyboard layouts (https://github.com/iotaledger/firefly/issues/1934), so this change should be tested with multiple keyboard layouts and OSes. I'm happy to help with that btw, but it may take a bit more time and could delay this PR if it's included in it

Comment (rajivshah3) [reply packages/shared/components/inputs/Pin.svelte:67]: IMO it should be in a separate PR because it can affect the functionality of the PIN component (making it potentially user-facing, and no longer considered a chore). We've seen reports of issues with other keyboard layouts (https://github.com/iotaledger/firefly/issues/1934), so this change should be tested with multiple keyboard layouts and OSes. I'm happy to help with that btw, but it may take a bit more time and could delay this PR if it's included in it

Comment (Tuditi) [reply packages/shared/components/inputs/Pin.svelte]: I cherry-picked the two relevant commits and opened a PR here: https://github.com/iotaledger/firefly/pull/3360.
Probably need to rebase once the PR has been tested. Can you test the different layouts and OS'es?

File: packages/shared/lib/profile.ts (hunk 3 above)
Comment (Tuditi) [reply packages/shared/lib/profile.ts:1]: I removed that file
